flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClientActor.java

```diff
@@ -59,11 +87,50 @@ public JobClientActor(
 		this.leaderSessionID = Preconditions.checkNotNull(leaderSessionID, "The leader session ID option must not be null.");
 
 		this.sysoutUpdates = sysoutUpdates;
+		// set this to -1 to indicate the job hasn't been created yet.
+		this.currentJobCreatedAt = -1;
 	}
 	
 	@Override
 	protected void handleMessage(Object message) {
 		
+		// ======= Job status messages on regular intervals ==============
+		if(message instanceof JobManagerMessages.CurrentJobStatus){
+			JobStatus statusReport = ((JobManagerMessages.CurrentJobStatus) message).status();
+			long timeDiff;
+			switch(statusReport){
+				case RUNNING:
+					// Vincent, we happy?
+					this.currentJobCreatedAt = -1;
+					break;
+				case FINISHED:
+					// Yeah! We happy!
+					this.currentJobCreatedAt = -1;
+					break;
+				case CREATED:
```

Comment:
For simple jobs, Flink supports queued scheduling. Thus, it might be the case that one job stays queued up for quite some time until its execution gets started.